# Importing the Libraries

In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────────────────────────────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


# Reading the CSV Files

In [2]:
daily_activity <- read_csv("Fitabase Data 3.12.16-4.11.16/dailyActivity_merged.csv")

Rows: 457 Columns: 15
── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
chr  (1): ActivityDate
dbl (14): Id, TotalSteps, TotalDistance, TrackerDistance, LoggedActivitiesDi...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [3]:
heart_rate_seconds <- read_csv("Fitabase Data 3.12.16-4.11.16/heartrate_seconds_merged.csv")

Rows: 1154681 Columns: 3
── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
chr (1): Time
dbl (2): Id, Value

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [4]:
hourly_calories <- read_csv("Fitabase Data 3.12.16-4.11.16/hourlyCalories_merged.csv")

Rows: 24084 Columns: 3
── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
chr (1): ActivityHour
dbl (2): Id, Calories

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [5]:
hourly_intensities <- read_csv("Fitabase Data 3.12.16-4.11.16/hourlyIntensities_merged.csv")

Rows: 24084 Columns: 4
── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
chr (1): ActivityHour
dbl (3): Id, TotalIntensity, AverageIntensity

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [6]:
hourly_steps <- read_csv("Fitabase Data 3.12.16-4.11.16/hourlySteps_merged.csv")

Rows: 24084 Columns: 3
── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
chr (1): ActivityHour
dbl (2): Id, StepTotal

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [7]:
sleep <- read_csv("Fitabase Data 3.12.16-4.11.16/minuteSleep_merged.csv")

Rows: 198559 Columns: 4
── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
chr (1): date
dbl (3): Id, value, logId

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [8]:
weight_info <- read_csv("Fitabase Data 3.12.16-4.11.16/weightLogInfo_merged.csv")

Rows: 33 Columns: 8
── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
chr (1): Date
dbl (6): Id, WeightKg, WeightPounds, Fat, BMI, LogId
lgl (1): IsManualReport

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


# Perviewing the Imported Datasets

In [9]:
head(daily_activity)
sum(is.na(daily_activity)) # finds the count of missing values 
length(unique(daily_activity$Id)) # finds unique Ids in the data set
sum(duplicated(daily_activity))

Id         ActivityDate TotalSteps TotalDistance TrackerDistance
1 1503960366 3/25/2016    11004       7.11          7.11          
2 1503960366 3/26/2016    17609      11.55         11.55          
3 1503960366 3/27/2016    12736       8.53          8.53          
4 1503960366 3/28/2016    13231       8.93          8.93          
5 1503960366 3/29/2016    12041       7.85          7.85          
6 1503960366 3/30/2016    10970       7.16          7.16          
  LoggedActivitiesDistance VeryActiveDistance ModeratelyActiveDistance
1 0                        2.57               0.46                    
2 0                        6.92               0.73                    
3 0                        4.66               0.16                    
4 0                        3.19               0.79                    
5 0                        2.16               1.09                    
6 0                        2.36               0.51                    
  LightActiveDistance SedentaryActiveDistance VeryActiveMinutes
1 4.07                0                       33               
2 3.91                0                       89               
3 3.71                0                       56               
4 4.95                0                       39               
5 4.61                0                       28               
6 4.29                0                       30               
  FairlyActiveMinutes LightlyActiveMinutes SedentaryMinutes Calories
1 12                  205                   804             1819    
2 17                  274                   588             2154    
3  5                  268                   605             1944    
4 20                  224                  1080             1932    
5 28                  243                   763             1886    
6 13                  223                  1174             1820

[1] 0

[1] 35

[1] 0

In [10]:
head(heart_rate_seconds)
sum(is.na(heart_rate_seconds))
length(unique(heart_rate_seconds$Id))
sum(duplicated(heart_rate_seconds))

Id         Time                Value
1 2022484408 4/1/2016 7:54:00 AM  93  
2 2022484408 4/1/2016 7:54:05 AM  91  
3 2022484408 4/1/2016 7:54:10 AM  96  
4 2022484408 4/1/2016 7:54:15 AM  98  
5 2022484408 4/1/2016 7:54:20 AM 100  
6 2022484408 4/1/2016 7:54:25 AM 101

[1] 0

[1] 14

[1] 0

In [11]:
head(hourly_calories)
sum(is.na(hourly_calories))
length(unique(hourly_calories$Id))
sum(duplicated(hourly_calories))

Id         ActivityHour          Calories
1 1503960366 3/12/2016 12:00:00 AM 48      
2 1503960366 3/12/2016 1:00:00 AM  48      
3 1503960366 3/12/2016 2:00:00 AM  48      
4 1503960366 3/12/2016 3:00:00 AM  48      
5 1503960366 3/12/2016 4:00:00 AM  48      
6 1503960366 3/12/2016 5:00:00 AM  48

[1] 0

[1] 34

[1] 0

In [12]:
head(hourly_intensities)
sum(is.na(hourly_intensities))
length(unique(hourly_intensities$Id))
sum(duplicated(hourly_intensities))

Id         ActivityHour          TotalIntensity AverageIntensity
1 1503960366 3/12/2016 12:00:00 AM 0              0               
2 1503960366 3/12/2016 1:00:00 AM  0              0               
3 1503960366 3/12/2016 2:00:00 AM  0              0               
4 1503960366 3/12/2016 3:00:00 AM  0              0               
5 1503960366 3/12/2016 4:00:00 AM  0              0               
6 1503960366 3/12/2016 5:00:00 AM  0              0

[1] 0

[1] 34

[1] 0

In [13]:
head(hourly_steps)
sum(is.na(hourly_steps))
length(unique(hourly_steps$Id))
sum(duplicated(hourly_steps))

Id         ActivityHour          StepTotal
1 1503960366 3/12/2016 12:00:00 AM 0        
2 1503960366 3/12/2016 1:00:00 AM  0        
3 1503960366 3/12/2016 2:00:00 AM  0        
4 1503960366 3/12/2016 3:00:00 AM  0        
5 1503960366 3/12/2016 4:00:00 AM  0        
6 1503960366 3/12/2016 5:00:00 AM  0

[1] 0

[1] 34

[1] 0

In [14]:
head(sleep)
sum(is.na(sleep))
length(unique(sleep$Id))
sum(duplicated(sleep)) # number of duplicates
sleep <- unique(sleep) # removing duplicate rows
sum(duplicated(sleep)) # to verify that duplicates have been removed or not

Id         date                 value logId      
1 1503960366 3/13/2016 2:39:30 AM 1     11114919637
2 1503960366 3/13/2016 2:40:30 AM 1     11114919637
3 1503960366 3/13/2016 2:41:30 AM 1     11114919637
4 1503960366 3/13/2016 2:42:30 AM 1     11114919637
5 1503960366 3/13/2016 2:43:30 AM 1     11114919637
6 1503960366 3/13/2016 2:44:30 AM 1     11114919637

[1] 0

[1] 23

[1] 525

[1] 0

In [15]:
head(weight_info)
sum(is.na(weight_info))
length(unique(weight_info$Id))
sum(duplicated(weight_info))

Id         Date                 WeightKg WeightPounds Fat BMI  
1 1503960366 4/5/2016 11:59:59 PM  53.3    117.5064     22  22.97
2 1927972279 4/10/2016 6:33:26 PM 129.6    285.7191     NA  46.17
3 2347167796 4/3/2016 11:59:59 PM  63.4    139.7731     10  24.77
4 2873212765 4/6/2016 11:59:59 PM  56.7    125.0021     NA  21.45
5 2873212765 4/7/2016 11:59:59 PM  57.2    126.1044     NA  21.65
6 2891001357 4/5/2016 11:59:59 PM  88.4    194.8886     NA  25.03
  IsManualReport LogId       
1  TRUE          1.459901e+12
2 FALSE          1.460313e+12
3  TRUE          1.459728e+12
4  TRUE          1.459987e+12
5  TRUE          1.460074e+12
6  TRUE          1.459901e+12

[1] 31

[1] 11

[1] 0

After the previewing of datasets, we conclude the followings:
- There are no missing values in selected data sets except the weight_info data frame
- The datasets are inconsistent as number of participants in each dataset is not same (Id column)
- A transformation is required for time/date column
- Duplicates were identified in the sleep data frame which are removed.

# Transformation of Date/Time
the date/time column for all datasets is character datatype. We will convert it into date/time data type in R

In [16]:
daily_activity$ActivityDate <- mdy(daily_activity$ActivityDate)

In [17]:
head(daily_activity)

Id         ActivityDate TotalSteps TotalDistance TrackerDistance
1 1503960366 2016-03-25   11004       7.11          7.11          
2 1503960366 2016-03-26   17609      11.55         11.55          
3 1503960366 2016-03-27   12736       8.53          8.53          
4 1503960366 2016-03-28   13231       8.93          8.93          
5 1503960366 2016-03-29   12041       7.85          7.85          
6 1503960366 2016-03-30   10970       7.16          7.16          
  LoggedActivitiesDistance VeryActiveDistance ModeratelyActiveDistance
1 0                        2.57               0.46                    
2 0                        6.92               0.73                    
3 0                        4.66               0.16                    
4 0                        3.19               0.79                    
5 0                        2.16               1.09                    
6 0                        2.36               0.51                    
  LightActiveDistance SedentaryActiveDistance VeryActiveMinutes
1 4.07                0                       33               
2 3.91                0                       89               
3 3.71                0                       56               
4 4.95                0                       39               
5 4.61                0                       28               
6 4.29                0                       30               
  FairlyActiveMinutes LightlyActiveMinutes SedentaryMinutes Calories
1 12                  205                   804             1819    
2 17                  274                   588             2154    
3  5                  268                   605             1944    
4 20                  224                  1080             1932    
5 28                  243                   763             1886    
6 13                  223                  1174             1820

In [18]:
heart_rate_seconds$Time <- mdy_hms(heart_rate_seconds$Time,tz = Sys.timezone())

In [19]:
head(heart_rate_seconds)

Id         Time                Value
1 2022484408 2016-04-01 07:54:00  93  
2 2022484408 2016-04-01 07:54:05  91  
3 2022484408 2016-04-01 07:54:10  96  
4 2022484408 2016-04-01 07:54:15  98  
5 2022484408 2016-04-01 07:54:20 100  
6 2022484408 2016-04-01 07:54:25 101

In [20]:
hourly_calories$ActivityHour <- mdy_hms(hourly_calories$ActivityHour, tz = Sys.timezone())

In [21]:
head(hourly_calories)

Id         ActivityHour        Calories
1 1503960366 2016-03-12 00:00:00 48      
2 1503960366 2016-03-12 01:00:00 48      
3 1503960366 2016-03-12 02:00:00 48      
4 1503960366 2016-03-12 03:00:00 48      
5 1503960366 2016-03-12 04:00:00 48      
6 1503960366 2016-03-12 05:00:00 48

In [22]:
hourly_intensities$ActivityHour <- mdy_hms(hourly_intensities$ActivityHour, tz = Sys.timezone())

In [23]:
head(hourly_intensities)

Id         ActivityHour        TotalIntensity AverageIntensity
1 1503960366 2016-03-12 00:00:00 0              0               
2 1503960366 2016-03-12 01:00:00 0              0               
3 1503960366 2016-03-12 02:00:00 0              0               
4 1503960366 2016-03-12 03:00:00 0              0               
5 1503960366 2016-03-12 04:00:00 0              0               
6 1503960366 2016-03-12 05:00:00 0              0

In [24]:
hourly_steps$ActivityHour <- mdy_hms(hourly_steps$ActivityHour, tz = Sys.timezone())

In [25]:
head(hourly_steps)

Id         ActivityHour        StepTotal
1 1503960366 2016-03-12 00:00:00 0        
2 1503960366 2016-03-12 01:00:00 0        
3 1503960366 2016-03-12 02:00:00 0        
4 1503960366 2016-03-12 03:00:00 0        
5 1503960366 2016-03-12 04:00:00 0        
6 1503960366 2016-03-12 05:00:00 0

In [26]:
sleep$date <- mdy_hms(sleep$date, tz = Sys.timezone())

In [27]:
head(sleep)

Id         date                value logId      
1 1503960366 2016-03-13 02:39:30 1     11114919637
2 1503960366 2016-03-13 02:40:30 1     11114919637
3 1503960366 2016-03-13 02:41:30 1     11114919637
4 1503960366 2016-03-13 02:42:30 1     11114919637
5 1503960366 2016-03-13 02:43:30 1     11114919637
6 1503960366 2016-03-13 02:44:30 1     11114919637

In [28]:
weight_info$Date <- mdy_hms(weight_info$Date, tz = Sys.timezone())

In [29]:
head(weight_info)

Id         Date                WeightKg WeightPounds Fat BMI   IsManualReport
1 1503960366 2016-04-05 23:59:59  53.3    117.5064     22  22.97  TRUE         
2 1927972279 2016-04-10 18:33:26 129.6    285.7191     NA  46.17 FALSE         
3 2347167796 2016-04-03 23:59:59  63.4    139.7731     10  24.77  TRUE         
4 2873212765 2016-04-06 23:59:59  56.7    125.0021     NA  21.45  TRUE         
5 2873212765 2016-04-07 23:59:59  57.2    126.1044     NA  21.65  TRUE         
6 2891001357 2016-04-05 23:59:59  88.4    194.8886     NA  25.03  TRUE         
  LogId       
1 1.459901e+12
2 1.460313e+12
3 1.459728e+12
4 1.459987e+12
5 1.460074e+12
6 1.459901e+12

# Analysis
we will uncover the general statistics of data and compare the results with appropriate data.

In [32]:
daily_activity %>%
    select(FairlyActiveMinutes) %>%
    summary()


 FairlyActiveMinutes
 Min.   :  0.00     
 1st Qu.:  0.00     
 Median :  1.00     
 Mean   : 13.07     
 3rd Qu.: 16.00     
 Max.   :660.00     

The average of FairlyActiveMinutes is 13.07 in a day while WHO recommends at least 21 to 42 minutes of moderate aerobic activity per day \
Reference : https://www.who.int/europe/publications/i/item/9789240014886

Now we examine FairlyActiveMinutes for each respondant.

In [50]:
daily_activity %>%
    group_by (Id) %>%
    summarise(Average_FairlyActiveMinutes = mean(FairlyActiveMinutes)) %>%
    filter (Average_FairlyActiveMinutes >= 21) %>% # 21 minutes are minimum moderate aerobic activity per day
    nrow()

[1] 8

It is important to mention that only 8 (23 %) respondents out of 35 are involved physical activity as per WHO guidelines.\
Also this information holds potential for enhancing Bellabeat's marketing strategy by encouraging customers toward increased activity over sedentary behavior. Moreover, the Bellabeat app could utilize timed notifications to prompt users towards activity rather than remaining sedentary.

In [51]:
daily_activity %>%
    select (TotalSteps) %>%
    summary()

   TotalSteps   
 Min.   :    0  
 1st Qu.: 1988  
 Median : 5986  
 Mean   : 6547  
 3rd Qu.:10198  
 Max.   :28497  

According to a study, 10,000 steps per day is a reasonable target for healthy adults.\
Reference : https://www.healthline.com/health/how-many-steps-a-day#how-many-steps-per-day \
The mean fo TotalSteps for all respondents is 6547 which is quite low. \
Bellabeat could utilize this data to incentivize customers who achieve the 10,000-step target with rewards. This initiative could create mutual benefits for both Bellabeat and its customers, fostering a win-win scenario.